### ISSUE TITLE
Pause the action queue when attacking

### ISSUE THREAD

**Issue by BrunoMine** [hkzorman/advanced_npc#33]
If npc is in attack state (`self.state == "attack"`), it must interrompe tasks.

**Comment by hkzorman** [comment]
I agree there's still not enough integration between both APIs (advanced_npc and mobs_redo).
I plan to make changes on how the `on_step` works, because:
* The wandering in mobs_redo is too clumsy and make NPCs go into wrong directions, keep walking when they are colliding, and they overall look crazy. I want to change this to a more natural wandering, and actually, much more limited
* While the action/task queue is executed, NPCs can be punched and they don't react
* NPCs attacks are too deadly. They will attack until you are killed or out of sight. Maybe they shouldn't be too aggressive (maybe this can be configured as well), specially if you are already into some relationship with them
* The issue you just mentioned

So yes, I will definitely address this.

**Comment by BrunoMine** [comment]
I made some adjustments in my project to hide the problem, but I look forward to correcting it.

**Comment by BrunoMine** [comment]
Has this problem been fixed? It seems I'm still seeing this occur.
